src/test/java/io/wgj/ctci/StackTest.java

```diff
@@ -0,0 +1,78 @@
+package io.wgj.ctci;
+
+import static org.junit.Assert.*;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
+import java.util.stream.IntStream;
+
+public class StackTest {
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testStackOneInt() {
+        Stack s = new Stack();
+        s.push(1);
+        int r = s.pop();
+        assertEquals(1, r);
+    }
+
+    @Test
+    public void testStackMultipleInts() {
+        ArrayList<Integer> expected = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            expected.add(i);
+        }
+
+        Stack s = new Stack();
+        for (int e : expected) {
+            s.push(e);
+        }
+
+        Collections.reverse(expected);
+        for (int e : expected) {
+            assertEquals(e, s.pop());
+        }
+    }
+
+    @Test
+    public void testStackRandomInts() {
+        Random r = new Random();
```

Comment:
This is more of a fuzzer than a test.  If it did find a bug, how would you be able to determine the seed that reproduces the problem?  The best tests do *exactly* the same thing every time.  They can have randomness, but it better be there for a good reason.  As-is this test seems basically identical to testStackMultipleInts except it could be more flaky because of the randomness.

Reply:
Fixed in cd6dc83cc0daf94827e214ba2a7c26f33c9aef00